### PR TITLE
modified task_agent.py to handle logging messages

### PR DESCRIPTION
**Pull request by johnnykfeng**
Add `self.logger` as a attribute in `TaskAgent` class, used the logger attribute to store thought process of agent

logger is updated inside the `run` method

modified `process_chat_history` method to remove the verbose text in previous messages

### CODE REVIEW

File: apps/slackbot/task_agent.py [hunk 1]
```diff
@@ -119,6 +121,14 @@ def run(self, task: str) -> str:
                 user_input=user_input,
             )
             print("reply:", assistant_reply)
+            # added by JF
+            try:
+                reply_json = json.loads(assistant_reply)
+                logger_step['reply'] = reply_json
+            except:
```
Comment (20001LastOrder) [inline apps/slackbot/task_agent.py:128]: Using a generic except is not a very good practice. After some search I found the `json.loads` function thoughts [`JSONDecodeError`](https://docs.python.org/3/library/json.html#json.JSONDecodeError) if the input is not a valid JSON. So we can catch that specific error instead like

``` python
try:
    json.loads(...)
except json.JSONDecoderError as e:
    ...
```

File: apps/slackbot/task_agent.py [hunk 2]
```diff
@@ -204,6 +214,12 @@ def process_chat_history(self, messages: List[dict]) -> List[BaseMessage]:
             message_cls = AIMessage if message['user'] == self.ai_id else HumanMessage
             # replace the at in the message with the name of the bot
             text = message['text'].replace(f'@{self.ai_id}', f'@{self.ai_name}')
+            # added by JF
+            try:
+                text = text.split("#verbose", 1)[0]  # remove everything after #verbose
```
Comment (20001LastOrder) [inline apps/slackbot/task_agent.py:219]: What are these things for? I'm not seeing anything else about verbose in this PR.

Comment (johnnykfeng) [reply apps/slackbot/task_agent.py:219]: I combine the final response and though process in this snippet, inside line 220 of `bolt_app.py`, which separates the thought process and response with `#verbose` in between. But I don't want to include the thought process in the message history, so I just add that to `process_chat_history` to clean the though process, i.e. everything after #verbose. 

`    if contains_verbose(query = question):
      print("Verbose mode is on")
      question = question.replace(f'@{ai_id}', f'@{ai_name}')
      question = question.replace('-verbose', '')
      response = task_agent.run(question)
      logger = task_agent.logger # logger is updated after running task_agent.run(question)
      verbose_response = response + ' \n #verbose \n ' + log_formatter(logger)
      return verbose_response`

Comment (20001LastOrder) [reply apps/slackbot/task_agent.py:219]: OK, I get it now. In this case, I don't think you need the `try` `except` block. If there is no `#verbose` in the text, then the result of the `split` function will just be a list of one item. For example:
``` python
a = "This is a test message with #verbose Let's say"
a.split('#verbose', 1) # result: ['This is a test message with ', " Let's say"]

b = "This is a test message with Let's say"
b.split('#verbose', 1) # result: ["This is a test message with Let's say"]
```

The `split` function works similarly, I think. Also in the future, you can combine the two PRs together (for example, merging them to the same branch of your repository) if they depend on each other :)

**Comment by johnnykfeng** [comment]
@20001LastOrder made a new commit based on suggestions, please check them out when you have time

**Review by 20001LastOrder — APPROVED** [review]
Looks good to me. Let's merge it and do some live testing!

**Pull request merged**